File: dashboard/src/utils/docsHref.ts
export const DOCS_MOUNT = '/help';

function resolveDocsLocale(locale?: string): string {
  if (locale) {
    return locale;
  }
  if (typeof localStorage !== 'undefined') {
    return localStorage.getItem('astrbot-locale') || 'zh-CN';
  }
  return 'zh-CN';
}

export function docsHref(path = '', locale?: string): string {
  const isEn = resolveDocsLocale(locale).toLowerCase().startsWith('en');
  const prefix = isEn ? `${DOCS_MOUNT}/en` : DOCS_MOUNT;
  const clean = path.replace(/^\/+/, '');
  if (!clean || clean === 'index.html') {
    return `${prefix}/`;
  }
  return `${prefix}/${clean}`;
}

export function configDocsHref(docs?: unknown, locale?: string): string {
  if (typeof docs !== 'string') {
    return '';
  }
  const clean = docs.trim().replace(/^\/+/, '');
  if (!clean || clean === 'index.html') {
    return '';
  }
  return docsHref(clean, locale);
}

File: astrbot/dashboard/docs_href.py
"""Same-origin documentation paths served under ``/help``."""

DOCS_MOUNT = "/help"


def docs_href(path: str = "", *, english: bool = False) -> str:
    """Return a WebUI documentation path.

    Args:
        path: Page path relative to the docs root, with or without a
            leading slash. An empty path returns the locale index.
        english: When True, prefix the English locale tree.

    Returns:
        An absolute path on the Dashboard origin, such as
        ``/help/faq.html`` or ``/help/en/faq.html``.
    """
    clean = path.strip().lstrip("/")
    prefix = f"{DOCS_MOUNT}/en" if english else DOCS_MOUNT
    if not clean:
        return f"{prefix}/"
    return f"{prefix}/{clean}"
